compiler/rustc_mir/src/const_eval/eval_queries.rs

```diff
@@ -248,18 +263,29 @@ pub fn eval_to_allocation_raw_provider<'tcx>(
     // computed. For a large percentage of constants that will already have succeeded. Only
     // associated constants of generic functions will fail due to not enough monomorphization
     // information being available.
-
-    // In case we fail in the `UserFacing` variant, we just do the real computation.
-    if key.param_env.reveal() == Reveal::All {
-        let mut key = key;
-        key.param_env = key.param_env.with_user_facing();
-        match tcx.eval_to_allocation_raw(key) {
-            // try again with reveal all as requested
```

Comment:
You lost this comment.

Also, the `Reveal` logic here is an exact duplicate of the one above, isn't it? By now it looks non-trivial enough to justify doing some refactoring to share it.